src/assets/all_assets_request.rs

```diff
@@ -107,9 +73,15 @@ impl AllAssetsRequest {
     /// # Returns
     /// The request object
     /// [AllAssetsRequest](struct.AllAssetsRequest.html)
-    pub fn set_asset_code(&mut self, asset_code: &str) -> &mut Self {
-        self.asset_code = Some(asset_code.to_owned());
-        self
+    pub fn set_asset_code(self, asset_code: &str) -> Result<AllAssetsRequest, String> {
+        if asset_code.len() > 12 {
+            return Err("asset_code must be 12 characters or less".to_string());
+        }
```

Comment:
These are still runtime errors. We could introduce an 'InvalidAssetRequest' state and return this if the asset_code > 12.